wp-includes/parser/class-wp-parser.php

```diff
@@ -26,6 +33,37 @@ public function parse() {
 		return false === $ast ? null : $ast;
 	}
 
+	/**
+	 * Parse the next AST from the input tokens.
+	 *
+	 * This method reads tokens from the input until an AST is successfully parsed.
```

Comment:
I'm lost reading this comment. It expresses code-level ideas one line at a time. Let's explain what this method does for the developer and include a brief usage example to illustrate the idea.